app/components/CreateQuestionContainer.tsx

```diff
@@ -40,27 +40,51 @@ export default function CreateQuestion({
   const questionBody = useRef<HTMLTextAreaElement | null>(null);
   const questionTitle = useRef<HTMLInputElement | null>(null);
 
-  const createQuestion = useContractWrite(
-    {
-      addressOrName: questionAPI.address,
-      contractInterface: questionAPI.abi,
+    // With the migration to 0.6+ wagmi, they recommend preparing the transaction and
```

Comment:
Thanks for the comments - we'll have to talk this through - feels very weird but understand the limitations `recklesslyUnprepared` reminds me of `dangerouslySetInnerHTML` in React - seems kinda silly 

I'm thinking for at least this part we might use the back end with plain ethers.js to call this functionality as we need to use the backend to post to IPFS and we can add true form validation and such

Reply:
Right, we really can't benefit from using this in this instance without rethinking UX and I'm not sure a 2 click process is worth the friction.

Reply:
Would this work?:

```tsx
const { config, isSuccess } = usePrepareContractWrite({
  addressOrName: questionAPI.address,
  contractInterface: questionAPI.abi,
  functionName: "createQuestion",
  args: network === "polygon" ? [fileUrl, BigNumber.from("10")] : [fileUrl],
  enabled: Boolean(fileUrl)
});
const createQuestion = useContractWrite(config);

useEffect(() => {
  if (isSuccess) {
    askQuestion();
  }
}, [isSuccess]);
```

I pulled the repo and ran this code locally, and it seemed to work fine for me.

Also, sorry to be snooping this PR, I was just searching for "wagmi" in GitHub search. 😂

Reply:
@jxom worked like a charm, can't tell you how much I appreciate you dropping in and helping me out!

I had been chipping away all day at these two issues and had a dependency conflict with 0.6.4 and react query initially so most of the trial and error with your new usePrepare hook was done in 0.6.0. Found the discussion on cache size and just rolled with reckless prep glossed over the enabled arg before fixing the dependency issue and getting back up to 0.6.4.

Feel free to stop by anytime jxom 😉 thank you again

Reply:
> Would this work?:
> 
> ```tsx
> const { config, isSuccess } = usePrepareContractWrite({
>   addressOrName: questionAPI.address,
>   contractInterface: questionAPI.abi,
>   functionName: "createQuestion",
>   args: network === "polygon" ? [fileUrl, BigNumber.from("10")] : [fileUrl],
>   enabled: Boolean(fileUrl)
> });
> const createQuestion = useContractWrite(config);
> 
> useEffect(() => {
>   if (isSuccess) {
>     askQuestion();
>   }
> }, [isSuccess]);
> ```
> 
> I pulled the repo and ran this code locally, and it seemed to work fine for me.
> 
> Also, sorry to be snooping this PR, I was just searching for "wagmi" in GitHub search. 😂

Your comments are much appreciated @jxom